### ISSUE TITLE
Trouble with too many newlines in rendered output

### ISSUE THREAD

**Issue by mariusk** [tj/haml.js#24]
When mixing pure text with invisible span elements (email obfuscation), it seems the way this module inserts newlines after each element inserts spaces where there should be none. I've attached a patch that remove those newlines below. Please note that the patch also includes an "include" filter which pulls in external haml/text fragments from (which could just as well be implemented outside of the core module, I know):

```
--- /root/src/haml.js/lib/haml.js       2010-05-21 19:33:50.000000000 +0000
+++ lib/haml.js 2010-05-21 19:28:14.000000000 +0000
@@ -77,7 +77,24 @@

   javascript: function(str) {
     return '<script type="text/javascript">\n//<![CDATA[\n' + str + '\n//]]></script>'
+  },
+
+  /**
+   * Include text/haml fragments from external files. There should probably
+   * be some config somehwere to say where/what the root, plus encoding and
+   * default options to haml render.
+   */
+
+  include: function(filename) {
+
+    var data = fs.readFileSync(filename, "utf8");
+    if (filename.match (new RegExp ('\\.haml$')) // RegExp should be set up once
+)
+      return haml.render(data, { filename: filename, cache: 1});
+    else
+      return data;
   }
+
 }

 /**
@@ -380,7 +397,8 @@
   get tag() {
     var tag = this.advance.val,
         selfClosing = !this.xml && exports.selfClosing.indexOf(tag) !== -1,
-        buf = ['"\\n<' + tag + this.attrs + (selfClosing ? '/>"' : '>"')]
+        //buf = ['"\\n<' + tag + this.attrs + (selfClosing ? '/>"' : '>"')]
+        buf = ['"<' + tag + this.attrs + (selfClosing ? '/>"' : '>"')]
     switch (this.peek.type) {
       case 'text':
         buf.push(this.text)
@@ -400,7 +418,8 @@
       case 'indent':
         buf.push(this.block)
     }
-    if (!selfClosing) buf.push('"</' + tag + '>\\n"')
+    //if (!selfClosing) buf.push('"</' + tag + '>\\n"')
+    if (!selfClosing) buf.push('"</' + tag + '>"')
     return buf.join(' + ')
   },
```

**Comment by tj** [comment]
this should really be two patches, and preferably not include old commented out code :p single patches will show the intention much better. Also I dont think a filter is the right place for "include"

**Comment by mariusk** [comment]
Agree. What would be the best way to send you patches (putting them here is a PITA). I may have a patch to add HEAD support as well.

**Comment by tj** [comment]
just fork the project, and then create a topic branch for fixes or whatever, 
I can pick and choose whatever to merge in

**Comment by mariusk** [comment]
Ok, I made an attempt. The patch for removing the newline is here:

http://github.com/mariusk/haml.js/commit/0de93a475fc813b09950ca6060d507a05c24a182

Haven't contributed patches through github before, so my apologies in advance if there are better ways.

**Comment by tj** [comment]
cool man i will merge that over

**Comment by tj** [comment]
Fixed issue with to many newlines in output. Closed by 8652c4f6702ee7ecfe985c82aad4ade46d2af980
